docs/releasing.md

```diff
@@ -1,80 +1,52 @@
+    * If doing a minimal bug-fix release, the changelog will only contain the changes you're intending to release.
     * Make sure all changelog entries have links back to their PR(s) if appropriate.
-    * For picking the new version number, we default to increasing the patch version number, but make our own judgement about whether a change is big enough to warrant a minor version bump. Common reasons for a minor bump could include:
-      * Making substantial new features available to all users. This can include lifting a feature flag.
-      * Breakage in compatibility with recent versions of the CLI.
-      * Minimum required version of VS Code is increased.
-      * New telemetry events are added.
-      * Deprecation or removal of commands.
-      * Accumulation of many changes, none of which are individually big enough to warrant a minor bump, but which together are. This does not include changes which are purely internal to the extension, such as refactoring, or which are only available behind a feature flag.
-1. Double-check that the node version we're using matches the one used for VS Code. See the [Node.js version instructions](./node-version.md) for more information.
 1. Double-check that the extension `package.json` and `package-lock.json` have the version you intend to release. If you are doing a patch release (as opposed to minor or major version) this should already be correct.
-1. Create a PR for this release:
-    * This PR will contain any missing bits from steps 1, 2 and 3. Most of the time, this will just be updating `CHANGELOG.md` with today's date.
-    * Create a new branch for the release named after the new version. For example: `v1.3.6`
-    * Create a new commit with a message the same as the branch name.
-    * Create a PR for this branch.
-    * Wait for the PR to be merged into `main`
-1. Switch to `main` branch and pull latest changes
-1. Lock the `main` branch.
-    * Go to the [branch protection rules for the `main` branch](https://github.com/github/vscode-codeql/settings/branch_protection_rules/16447115)
-    * Select "Lock branch"
-    * Click "Save changes"
-1. Ensure that no PRs have been merged since the release PR that you merged. If there were, you might need to unlock `main` temporarily and update the CHANGELOG again.
-1. Build the extension `npm run build` and install it on your VS Code using "Install from VSIX".
+1. Commit any changes made during steps 4 and 5 with a commit message the same as the branch name (e.g. `v1.3.6`).
+1. Open a PR for this release.
+    * The PR diff should contain:
+        * Any missing bits from steps 4 and 5. Most of the time, this will just be updating `CHANGELOG.md` with today's date.
+        * If releasing from a branch other than `main`, this PR will also contain the extension changes being released.
+1. Build the extension using `npm run build` and install it on your VS Code using "Install from VSIX".
```

Comment:
Do we no longer need to lock the branch we're releasing from?

Reply:
I don't believe so. Please do say if you think I'm wrong.

One branch to consider is the release branch. I don't believe there's a need to lock this branch as it's highly unlikely for another developer to push anything to it while you're doing the release.

I also don't believe it's necessary to lock the `main` branch. It shouldn't be a problem if anything is pushed to main while the release prep is happening. Those changes won't be included in the current release but that's ok, and they'll be included in the next release instead.

Thinking about this did make me realise that there could be conflicts in the changelog file merging the release PR, if any new PRs have been merged. But this solvable, and in this case the new entries should be placed at the top, above the section for the current release, so they will be marked as unreleased and ready to be included in the next release. I'll include a note in the instructions about this.

I'd have to check to make sure but I expect the workflow that updates the changelog after a release should continue to work even if more things have been pushed, so long as people stick to the right format.

Reply:
Ah yeah we're always releasing from a branch now so it makes sense not to have locks! Agree about conflicts but that's manageable.